src/main/scala/exercises/BalancedBrackets.scala

```diff
@@ -0,0 +1,69 @@
+package exercises
+
+import scala.annotation.tailrec
+
+class BalancedBrackets {
```

Comment:
Any reason for making this a class and also having its companion object below extend it?